sockeye/init_embedding.py

```diff
@@ -104,21 +111,26 @@ def main():
     if len(args.embeddings) != len(args.vocabularies_in) or \
        len(args.embeddings) != len(args.vocabularies_out) or \
        len(args.embeddings) != len(args.names):
-           logger.error("Exactly the same number of 'input embedding weights', 'input vocabularies', "
+           logger.error("Exactly the same number of 'input weights', 'input vocabularies', "
                         "'output vocabularies' and 'Sockeye parameter names' should be provided.")
            sys.exit(1)
 
     params = {} # type: Dict[str, mx.nd.NDArray]
-    for embed_file, vocab_in_file, vocab_out_file, name in zip(args.embeddings, args.vocabularies_in, \
+    for weight_file, vocab_in_file, vocab_out_file, name in zip(args.embeddings, args.vocabularies_in, \
                                                                args.vocabularies_out, args.names):
-        logger.info('Loading input embedding weight: %s', embed_file)
-        embed = np.load(embed_file)
+        logger.info('Loading input weight: %s', weight_file)
+        if weight_file.endswith(".npy"):
+            weight = np.load(weight_file)
+        elif weight_file.endswith(".npz"):
+            weight = np.load(weight_file)[name]
```

Comment:
for both the npz and the mx.nd.load potentially multiple weights come from the same file. Would it make sense to either: load the file only once and then pull out the different weights or restructure the CLI slightly, say:
```
--embeddings emb.npy params.best --names emb weight1:weight2
```
so that for each file one basically accepts a list of names (separated by `:` or some other delimiter).

Reply:
I cached loaded files instead of changing the CLI. If weight-file->weight-name supports 1-N mappings, then vocab-file->weight-name is also supposed to support 1-N mappings, which makes conditions too complicated. Let's only use 1-1 mappings since it's clear.

Reply:
sounds good.